content/en/blog/_posts/2025-04-21-kubernetes-network-evolution/index.md

```diff
@@ -0,0 +1,71 @@
+
+### Contributor Summit Paris (KubeCon 2024): Towards Modular Network Drivers
+
+To address the continuing challenges and ensure a clearer path forward, SIG Network Tech Leads proposed a more modular development model called Kubernetes Network Drivers. This proposal was presented at the [Future of k8s networking](https://kcseu2024.sched.com/event/1aOqO/future-of-k8s-networking?iframe=no&w=100%&sidebar=yes&bg=no) session during the Contributor Summit in Paris at KubeCon 2024, and it incorporated feedback from earlier discussions.
+
+{{< figure src="kubernetes_network_driver.png" alt="kubernetes network drivers" >}}
+
+This approach aimed to break down the complex problem of Kubernetes networking into smaller, more manageable components for integration, suggesting a move away from monolithic solutions towards a more composable architecture. This initiative seeks to provide a clearer framework for developing and integrating advanced networking functionalities into Kubernetes.
+
+### DRA, NRI, and Runtime Specification Changes: Foundational Building Blocks
+
+A significant advancement in supporting more complex and declarative networking was the integration of Dynamic Resource Allocation (DRA). Building upon DRA's ongoing work to abstract hardware in core Kubernetes components, the Kubernetes Network Driver model offered a streamlined path forward. By moving the ResourceClaim to the pod level, network interfaces could be treated as first-class resources, simplifying the user experience to a basic request of "give me a network."
+
+To facilitate the integration of network resources with DRA, the Node Resource Interface (NRI), which has roots in the CNI model but evolved towards a gRPC service model, was enhanced to support network development functions ([https://github.com/containerd/nri/pull/119](https://github.com/containerd/nri/pull/119)). Furthermore, a significant advancement occurred at the OCI runtime level. A proposal to offload the Network Interfaces plumbing to the runtime (`runc`) was approved ([https://github.com/opencontainers/runtime-spec/pull/1271](https://github.com/opencontainers/runtime-spec/pull/1271)) during Kubecon London 2025. This change, driven by observed inefficiencies and complexities in managing network interfaces using CNI plugins for simple network interfaces operations, allows the OCI runtime specification to natively support the declarative attachment of network interfaces to containers. This shift will simplify integrations and improve efficiency, particularly for workloads heavily reliant on network interfaces like AI/ML.
```

Comment:
Need to explain that NRI is a CRI thing.

"a significant advancement" should be "a significant advancement toward using DRA for networking". And then the following sentence makes no sense to anyone who hasn't been closely following this already. You need to explain what specific thing a DRA (or NRI) plugin can do now that it couldn't do before.

Reply:
you can use NRI without DRA for networking , see the two plugins I added as example on the repo https://github.com/containerd/nri/pulls?q=is%3Apr+author%3Aaojea+is%3Aclosed ,